sentry-uitest/sentry-uitest-android/src/androidTest/java/io/sentry/uitest/android/mockservers/EnvelopeAsserter.kt

```diff
@@ -0,0 +1,35 @@
+package io.sentry.uitest.android.mockservers
+
+import io.sentry.Sentry
+import io.sentry.SentryEnvelope
+import okhttp3.mockwebserver.MockResponse
+
+/**
+ * Class to make assertions on an envelope caught by [MockRelay].
+ * It contains the sent envelope and the returned response, too.
+ */
+class EnvelopeAsserter(val envelope: SentryEnvelope, val response: MockResponse) {
+    /** List of items to assert. */
+    private val unassertedItems = envelope.items.toMutableList()
+
+    /**
+     * Asserts an envelope item that can be deserialized as an instance of [clazz] exists and returns the first one.
+     * The asserted item is then removed from internal list of unasserted items.
+     */
+    fun <T> assertItem(clazz: Class<T>): T {
```

Comment:
We have the project `sentry-test-support` that does something similar, maybe move this in there, and avoid using `Sentry.getCurrentHub()`, you could simple create a new `JsonSerializer`

Reply:
```suggestion
    inline fun <reified T> assertItem(): T {
```
Also, since we are in Kotlin, can we make it inline+reified, so we can use it like `assertItem<SentryEvent>()` and use `T::class.java` instead of `clazz`